sled-storage/src/manager.rs

```diff
@@ -539,21 +581,37 @@ impl StorageManager {
             }
         }
 
+        // We want to add all U.2 disks concurrently rather than sequentially.
+        // Instead of directly calling `self.add_disk` in a loop or duplicating
+        // the logic in `add_queued_disks`, we queue up all the U.2 disks to
+        // add and then call `add_queued_disks` directly if the storage key
+        // manager is ready.
         for raw_disk in raw_disks {
             let disk_id = raw_disk.identity().clone();
-            match self.add_disk(raw_disk).await {
-                Ok(AddDiskResult::DiskInserted) => should_update = true,
-                Ok(_) => (),
-                Err(err) => {
-                    warn!(
-                        self.log,
-                        "Failed to add disk to storage resources: {err}";
-                        "disk_id" => ?disk_id
-                    );
+            match raw_disk.variant() {
```

Comment:
Just to confirm - have you tested an "unencrypted" -> "encrypted" migration with this PR? I unfortunately didn't really have automated tests for that process